src/api/types/TypeRegistrySpec.js

```diff
@@ -0,0 +1,35 @@
+define(['./TypeRegistry', './Type'], function (TypeRegistry, Type) {
+    describe('The Type API', function () {
+        var typeregistry;
+
+        beforeEach(function () {
+            typeregistry = new TypeRegistry ();
+            typeregistry.addType('testtype', {
+                name: 'Test Type',
+                description: 'This is a test type.',
+                creatable: true
+            });
+        });
+        it('new types are registered successfully', function () {
+            expect(typeregistry.types.testtype.definition.name).toBe('Test Type');
+        });
+
+        it('type registry contains new keys', function () {
+            expect(typeregistry.listKeys ()).toContain('testtype');
+        });
+
+        it('types can be retrieved', function () {
+            expect(typeregistry.get('testtype').definition.description).toBe('This is a test type.');
+        });
+
```

Comment:
Can this spec be combined with the first one ('new types are registered successfully') since get() returns a successfully registered type?

Also, can getDescription() or getName() be used instead of reading those values from definition?

Reply:
I don't think the type has a getDescription method?